repos/system_upgrade/common/actors/scanpkgmanager/libraries/scanpkgmanager.py

```diff
@@ -28,5 +33,46 @@ def get_etc_releasever():
     return releasever
 
 
+def get_config_contents(config_path):
+    if os.path.isfile(config_path):
+        with open(config_path, 'r') as config:
+            return config.read()
+    else:
+        return ''
+
+
+def get_proxy_if_set(manager_config_path):
+    """
+    Get proxy address from specified package manager config.
+
+    :param str manager_config_path: path to a package manager config
+    :returns: proxy address or None when not set
+    :rtype: str
+    """
+
+    config = get_config_contents(manager_config_path)
+    proxy_match = re.search('^proxy[ \t]*=', config, re.MULTILINE)
+
+    if not proxy_match:
+        return None
+
+    # boundaries of the proxy address
+    proxy_address_start = proxy_match.end()
+    proxy_address_end = config.find('\n', proxy_address_start)
+
+    proxy_address = config[proxy_address_start:]
+    if proxy_address_end != -1:
+        # the proxy setting is not on the last line of the config file
+        proxy_address = proxy_address[:proxy_address_end]
```

Comment:
Regular expression are very, however, they can be hard to maintain. The purpose of `re.search` is clear, however, the boundary calculation is a bit hard to think about. It is apparent, that the proxy configuration will be on one line - that's why you are searching for the closest `\n`. Therefore, the problem we are trying to solve can be stated as follows: _Find a line in the config that starts with `proxy<whitespaces>=`. If you have found such a line, return what stands on the right side of `=` (without possible whitespaces)_.

With this formulation, we could derive a simpler solution, one that does not require searching for newlines and indexing into the contents of the config file. Your solution at the moment works, which is the most important thing, however, simplicity (and thus maintainability) should be our aim, especially since this code will be likely there for a very long time.